refactor(benchmark): transform code before codegen

camc314 opened the pull request:
@overlookmotel suggested this [here](https://github.com/oxc-project/oxc/pull/7926#discussion_r1886023496).

overlookmotel commented:
I think this is a good change. Our `codegen_sourcemap` benchmark previously was unrealistic because the AST it printed had no modifications made to it after the parser. In real world, I assume you always modify the AST in some way before codegen. What would be the point of parsing a file and printing it again if nothing changes in between?

But @Boshen can you confirm you agree?

Boshen commented on tasks/benchmark/benches/codegen.rs:
```diff
@@ -1,28 +1,48 @@
 
         let mut group = criterion.benchmark_group("codegen");
-        group.bench_with_input(id.clone(), &ret.program, |b, program| {
+        group.bench_with_input(id.clone(), &program, |b, program| {
```
codegen should still use the untransformed AST for testing typescript codegen.

codegen_sourcemap should use the transformed AST.

Boshen commented:
### Merge activity

* **Dec 16, 7:51 AM EST**: The merge label '0-merge' was detected. This PR will be added to the [Graphite merge queue](https://app.graphite.dev/merges?org=oxc-project&repo=oxc) once it meets the requirements.

camc314 committed:
refactor(benchmark): transform code before codegen (#7934)

@overlookmotel suggested this [here](https://github.com/oxc-project/oxc/pull/7926#discussion_r1886023496).
